app/controllers/alaveteli_pro/stripe_webhooks_controller.rb

```diff
@@ -67,4 +68,33 @@ def notify_exception(error)
     end
   end
 
+  # ignore any that don't match our plan namespace
+  def filter_hooks
+    plans = []
+    case @stripe_event.data.object.object
+    when 'subscription'
+      plans = get_plan_names(@stripe_event.data.object.items)
+    when 'invoice'
+      plans = get_plan_names(@stripe_event.data.object.lines)
+    end
+
+    # ignore any plans that don't start with our namespace
+    plans.delete_if { |plan| !plan_matches_namespace(plan) }
+
+    if plans.empty?
+      # reject it and throw it away, the emails are just noise
+      render json: { error: 'Does not appear to be one of our plans' },
+             status: 401
```

Comment:
won't that result in us accepting another app's webhook? (genuine question, I have not got this straight in my head yet - it might clear "our" copy?)

I should probably dig into the docs some more

Reply:
It's fine for us to respond with a 2xx and do nothing because the other apps will setup their own webhook endpoints. If this was left as a 401 then Stripe would retries periodically, eventually fail and notify us by email.

Reply:
Cool, thanks!